server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/ClusterBalanceStatsTests.java

```diff
@@ -0,0 +1,210 @@
+                startedIndex("index-hot-3", 2.5, 6L, "node-hot-1", "node-hot-3"),
+                startedIndex("index-warm-1", 0.0, 12L, "node-warm-1", "node-warm-2"),
+                startedIndex("index-warm-2", 0.0, 18L, "node-warm-3")
+            )
+        );
+
+        var stats = ClusterBalanceStats.createFrom(clusterState, TEST_WRITE_LOAD_FORECASTER);
+
+        assertThat(
+            stats,
+            equalTo(
+                new ClusterBalanceStats(
+                    Map.of(
+                        DATA_HOT_NODE_ROLE.roleName(),
+                        new ClusterBalanceStats.TierBalanceStats(
+                            new ClusterBalanceStats.MetricStats(7.0, 2.0, 3.0, 7.0 / 3, stdDev(3.0, 2.0, 2.0)),
+                            new ClusterBalanceStats.MetricStats(21.0, 6.0, 8.5, 7.0, stdDev(6.0, 8.5, 6.5)),
+                            new ClusterBalanceStats.MetricStats(36.0, 10.0, 16.0, 12.0, stdDev(10.0, 10.0, 16.0))
+                        ),
+                        DATA_WARM_NODE_ROLE.roleName(),
+                        new ClusterBalanceStats.TierBalanceStats(
+                            new ClusterBalanceStats.MetricStats(3.0, 1.0, 1.0, 1.0, 0.0),
+                            new ClusterBalanceStats.MetricStats(0.0, 0.0, 0.0, 0.0, 0.0),
+                            new ClusterBalanceStats.MetricStats(42.0, 12.0, 18.0, 14.0, stdDev(12.0, 12.0, 18.0))
+                        )
+                    )
+                )
+            )
+        );
+    }
+
+    private static ClusterState createClusterState(List<DiscoveryNode> nodes, List<Tuple<IndexMetadata.Builder, String[]>> indices) {
+        var discoveryNodesBuilder = DiscoveryNodes.builder();
+        for (DiscoveryNode node : nodes) {
+            discoveryNodesBuilder.add(node);
+        }
+
+        var metadataBuilder = Metadata.builder();
+        var routingTableBuilder = RoutingTable.builder();
+        for (var index : indices) {
+            var indexMetadata = index.v1()
+                .settings(settings(Version.CURRENT))
+                .numberOfShards(index.v2().length)
+                .numberOfReplicas(0)
+                .build();
+            metadataBuilder.put(indexMetadata, false);
+            var indexRoutingTableBuilder = IndexRoutingTable.builder(indexMetadata.getIndex());
+            for (int shardId = 0; shardId < index.v2().length; shardId++) {
+                indexRoutingTableBuilder.addShard(
+                    newShardRouting(new ShardId(indexMetadata.getIndex(), shardId), index.v2()[shardId], true, STARTED)
+                );
+            }
+            routingTableBuilder.add(indexRoutingTableBuilder.build());
+        }
+
+        return ClusterState.builder(ClusterName.DEFAULT)
+            .nodes(discoveryNodesBuilder)
+            .metadata(metadataBuilder)
+            .routingTable(routingTableBuilder)
+            .build();
+    }
+
+    private static Tuple<IndexMetadata.Builder, String[]> startedIndex(
```

Comment:
Consider declaring a private record instead of `Tuple` which should be a more descriptive.